dowhy/causal_identifier/id_identifier.py

```diff
@@ -0,0 +1,284 @@
+from typing import List, Optional, Union
+
+import networkx as nx
+
+from dowhy.causal_graph import CausalGraph
+from dowhy.causal_identifier.identify_effect import EstimandType
+from dowhy.utils.api import parse_state
+from dowhy.utils.graph_operations import find_ancestor, find_c_components, induced_graph
+from dowhy.utils.ordered_set import OrderedSet
+
+
+class IDExpression:
+    """
+    Class for storing a causal estimand, as a result of the identification step using the ID algorithm.
+    The object stores a list of estimators(self._product) whose porduct must be obtained and a list of variables (self._sum) over which the product must be marginalized.
+    """
+
+    def __init__(self):
+        self._product = []
+        self._sum = []
+
+    def add_product(self, element):
```

Comment:
Can we add typings to these method signatures?

Reply:
added types to method signatures :)